common/src/com/thoughtworks/go/config/materials/Materials.java

```diff
@@ -140,9 +134,9 @@ public void cleanUp(File baseFolder, ConsoleOutputStreamConsumer consumer) {
         return allowed;
     }
 
-    boolean hasOneMaterialUseBaseFolder() {
+    public boolean hasOneMaterialUseBaseFolder() {
         for (Material material : this) {
```

Comment:
I see this being public only for testing. 
It's not used anywhere else other than this class. Can't we modify existing integration tests? Or better modify the clean up method to be testable.